bundles/model/org.eclipse.smarthome.model.lsp/src/main/java/org/eclipse/smarthome/model/lsp/internal/ModelServer.java

```diff
@@ -0,0 +1,101 @@
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.eclipse.lsp4j.launch.LSPLauncher;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.xtext.ide.server.LanguageServerImpl;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Guice;
+
+/**
+ * A service component exposing a Language Server via sockets.
+ *
+ * @author Simon Kaufmann - initial contribution and API.
+ *
+ */
+@Component(immediate = true)
+public class ModelServer {
+
+    private final Logger logger = LoggerFactory.getLogger(ModelServer.class);
+    private final int PORT = 5007;
+    private ServerSocket socket;
+
+    @Activate
+    public void activate() {
+        new Thread(() -> {
+            listen();
+        }, "Language Server").start();
+    }
+
+    @Deactivate
+    public void deactivate() {
+        try {
+            if (socket != null && !socket.isClosed()) {
+                socket.close();
+            }
+        } catch (IOException e) {
+            logger.error("Error shutting down the Language Server", e);
+        }
+    }
+
+    private void listen() {
+        try {
+            socket = new ServerSocket(PORT);
+            logger.info("Language Server running on port {}", PORT);
+            while (!socket.isClosed()) {
+                logger.debug("Going to wait for a client to connect");
+                try {
+                    Socket client = socket.accept();
+                    new Thread(() -> {
+                        handleConnection(client);
+                    }, "Client " + client.getRemoteSocketAddress()).start();
+                } catch (IOException e) {
+                    logger.error("Error accepting the client connection", e);
+                }
+            }
+        } catch (IOException e) {
+            logger.error("Error starting the Language Server", e);
+        }
+    }
+
+    private void handleConnection(final Socket client) {
+        logger.debug("Client {} connected", client.getRemoteSocketAddress());
+        try {
+
+            LanguageServerImpl languageServer = Guice.createInjector(new RuntimeServerModule())
+                    .getInstance(LanguageServerImpl.class);
+
+            Launcher<LanguageClient> launcher = LSPLauncher.createServerLauncher(languageServer,
+                    client.getInputStream(), client.getOutputStream());
+            languageServer.connect(launcher.getRemoteProxy());
+            Future<?> future = launcher.startListening();
+            future.get();
+        } catch (IOException e) {
+            logger.error("Error communicating with the LSP client", e);
```

Comment:
This is likely an "external" issue like a client that closes the connection. If we want to log it at all, we at least should not include the stacktrace here as it is no bug on our end.